### PR TITLE
Test signup page with cypress

### PR DESCRIPTION
**Pull request by brunomichetti**
- Test of signup elements with cypress
- Added link test to homepage

- TODO:
- Test successful signup

### CODE REVIEW

File: cypress/integration/signup-page-spec.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      .type('fakepassword')
+      .should('have.value', 'fakepassword');
+    cy.get('input[name="password2"]')
+      .type('fakepassword')
+      .should('have.value', 'fakepassword');
+    cy.get('#react-select-2-input').type('MALE{enter}', { force: true });
+    cy.get('#react-select-2-input').type('FEMALE{enter}', { force: true });
+    cy.get('#react-select-2-input').type('NOT SPECIFIED{enter}', {
+      force: true,
+    });
```
Comment (agch-dev) [inline cypress/integration/signup-page-spec.js]: what is the assertion here?

File: cypress/integration/signup-page-spec.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  });
+
+  it('Test empty inputs', () => {
+    cy.get('button[type="submit"]').click();
+    cy.get('.user-form__alert').then((alerts) => {
+      const nameAlert = alerts[0];
+      const emailAlert = alerts[1];
+      const passwordAlert = alerts[2];
+      const passwordConfirmAlert = alerts[3];
+      const genderAlert = alerts[4];
```
Comment (agch-dev) [inline cypress/integration/signup-page-spec.js]:
```suggestion
      const [nameAlert, emailAlert, passwordAlert, passwordConfirmAlert, genderAlert] = alerts;
```

**Review by agch-dev — COMMENTED** [review]
great job!!

As a general comment, I think you are doing too many assertions per `it` statement. Tests are here to help you, it is best to write them in a way that when one of them fails you can know at a glance what the problem is.

By clumping all the field assertions on one `it` statement, you would have to open the test runner, run that test and take a look at the screenshot to pinpoint which field is the one giving you trouble.

Although it uses a different library, taking a look at how we separate tests in the forms of the react-native-base can give you a better idea of what I am talking about. Here is a [link](https://github.com/rootstrap/react-native-base/blob/master/tests/components/SignUpForm.spec.js) to the SignUpForm.spec.js for example

File: cypress/integration/signup-page-spec.js (hunk 1 above)
Comment (brunomichetti) [reply cypress/integration/signup-page-spec.js]: That the selection in the react select is correct.
Do you know other way of test this?

File: cypress/integration/homepage-spec.js [hunk 3]
```diff
@@ -28,8 +28,7 @@ describe('Test home page', () => {
   it('Test empty inputs', () => {
     cy.get('button[type="submit"]').click();
     cy.get('.user-form__alert').then((alerts) => {
-      const emailAlert = alerts[0];
-      const passwordAlert = alerts[1];
+      const { emailAlert, passwordAlert } = alerts;
```
Comment (agch-dev) [inline cypress/integration/homepage-spec.js]: if `alerts` is still an array, it should be like this: `const [emailAlert, passwordAlert] = alerts;`

**Review by agch-dev — APPROVED** [review]

**Review by tamarasuarezrod — APPROVED** [review]

**Comment by brunomichetti** [comment]
> great job!!
> 
> As a general comment, I think you are doing too many assertions per `it` statement. Tests are here to help you, it is best to write them in a way that when one of them fails you can know at a glance what the problem is.
> 
> By clumping all the field assertions on one `it` statement, you would have to open the test runner, run that test and take a look at the screenshot to pinpoint which field is the one giving you trouble.
> 
> Although it uses a different library, taking a look at how we separate tests in the forms of the react-native-base can give you a better idea of what I am talking about. Here is a [link](https://github.com/rootstrap/react-native-base/blob/master/tests/components/SignUpForm.spec.js) to the SignUpForm.spec.js for example

Thanks! I'll do this refactor in another PR!

**Pull request merged**